SuperMAG

aburrell opened the pull request:
Added instrument support for SuperMAG.

aburrell commented:
We need a real email to create a test account for SuperMAG.

rstoneback commented:
Very nice. 

I took care of my simple comments since I know you are currently tied up.

I also just added initial support for usernames when downloading test data. Still running into a problem since tag='stations' isn't run first. 

I plan on trying to figure out why the code for downloading tags other than 'stations' isn't sorting things out. Looks like you should have it covered.

rstoneback commented:
Service email account is in process.

aburrell commented:
Why the change to write binary?  I'm pretty sure the files are all ascii...  Is there a way to get binary files?

rstoneback commented:
Hey Angeline,

That is where I stopped. Last bug to sort out. It is a python 2/3 issue. Without the binary option python 2 worked, 3 didn’t. My quick idea was a binary write but that didn’t work.  I’ve been delayed in sorting it out.

Cheers,
Russell

On Aug 20, 2018, at 8:58 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


Why the change to write binary? I'm pretty sure the files are all ascii... Is there a way to get binary files?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-414325677>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeeKXNhcX1ijx_Un5Cg7PyVYYoyEIks5uSsCXgaJpZM4VRgTW>.

aburrell commented on pysat/instruments/supermag_magnetometer.py:
```diff
@@ -812,7 +813,7 @@ def append_ascii_data(file_strings, tag):
     while i < len(out_lines) - 1:
         idates.append(i)
-        lsplit = re.split('\t+', out_lines[i])
+        lsplit = re.split('\t', out_lines[i])
```
Does this still work?  I recall there being an inexact number of tabs between entries.

rstoneback commented:
I believe it does.

The test suite for the last commits on that branch passed for python 2.

The last commit actually failed because Travis CI killed the build for being slow.
The last failures I had before pushing the binary write were,



======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer indices
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 694, in download
    file_fmt=file_fmt)
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 744, in download
    local_file.write(out_data)
TypeError: write() argument must be str, not bytes
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): supermag.jhuapl.edu:80<http://supermag.jhuapl.edu:80>
urllib3.connectionpool: DEBUG: http://supermag.jhuapl.edu:80 "POST /mag/lib/services/??user=rstoneback&service=stations&fmt=ascii&year=2009 HTTP/1.1" 200 14339
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 688, in download
    smag_stat.load(date=current_date)
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 879, in load
    self.data, meta = self._load_data(date=self.date, fid=self._fid)
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 622, in _load_data
    sat_id=self.sat_id, **self.kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 148, in load
    if isinstance(fnames, basestring):
NameError: name 'basestring' is not defined

======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer all
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 688, in download
    smag_stat.load(date=current_date)
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 879, in load
    self.data, meta = self._load_data(date=self.date, fid=self._fid)
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 622, in _load_data
    sat_id=self.sat_id, **self.kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 148, in load
    if isinstance(fnames, basestring):
NameError: name 'basestring' is not defined

======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer stations
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 744, in download
    local_file.write(out_data)
TypeError: write() argument must be str, not bytes



On Aug 20, 2018, at 9:07 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


@aburrell commented on this pull request.

________________________________

In pysat/instruments/supermag_magnetometer.py<https://github.com/rstoneback/pysat/pull/65#discussion_r211274816>:

> @@ -812,7 +813,7 @@ def append_ascii_data(file_strings, tag):
     i = ihead + 1
     while i < len(out_lines) - 1:
         idates.append(i)
-        lsplit = re.split('\t+', out_lines[i])
+        lsplit = re.split('\t', out_lines[i])


Does this still work? I recall there being an inexact number of tabs between entries.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#pullrequestreview-147660112>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeX1GTmqBrM4CfXC6AFMOSBMhEx_vks5uSsKygaJpZM4VRgTW>.

rstoneback commented:
According to GitHub it has been 27 days since I’ve done anything on SuperMag.

How’d that happen?


On Aug 20, 2018, at 9:07 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


@aburrell commented on this pull request.

________________________________

In pysat/instruments/supermag_magnetometer.py<https://github.com/rstoneback/pysat/pull/65#discussion_r211274816>:

> @@ -812,7 +813,7 @@ def append_ascii_data(file_strings, tag):
     i = ihead + 1
     while i < len(out_lines) - 1:
         idates.append(i)
-        lsplit = re.split('\t+', out_lines[i])
+        lsplit = re.split('\t', out_lines[i])


Does this still work? I recall there being an inexact number of tabs between entries.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#pullrequestreview-147660112>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeX1GTmqBrM4CfXC6AFMOSBMhEx_vks5uSsKygaJpZM4VRgTW>.

aburrell commented:
I think proposals and vacation happened.

I'm messing around with it now :-P  I really hate the way string treatment changed.

From: Russell Stoneback <notifications@github.com>
Reply-To: rstoneback/pysat <reply@reply.github.com>
Date: Monday, August 20, 2018 at 13:00
To: rstoneback/pysat <pysat@noreply.github.com>
Cc: "Burrell, Angeline" <angeline.burrell@nrl.navy.mil>, Mention <mention@noreply.github.com>
Subject: Re: [rstoneback/pysat] SuperMAG (#65)

According to GitHub it has been 27 days since I’ve done anything on SuperMag.

How’d that happen?


On Aug 20, 2018, at 9:07 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


@aburrell commented on this pull request.

________________________________

In pysat/instruments/supermag_magnetometer.py<https://github.com/rstoneback/pysat/pull/65#discussion_r211274816>:

> @@ -812,7 +813,7 @@ def append_ascii_data(file_strings, tag):
i = ihead + 1
while i < len(out_lines) - 1:
idates.append(i)
- lsplit = re.split('\t+', out_lines[i])
+ lsplit = re.split('\t', out_lines[i])


Does this still work? I recall there being an inexact number of tabs between entries.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#pullrequestreview-147660112>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeX1GTmqBrM4CfXC6AFMOSBMhEx_vks5uSsKygaJpZM4VRgTW>.



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-414388956>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AGuC_je1OgJ2KY5_bziqXPdUI3bA6y_xks5uSusXgaJpZM4VRgTW>.

aburrell commented:
Soooo none of the downloads are working on travis, and I don't know why.

rstoneback commented:
Thanks for the report. I will put it on my list.

I ran through the unit tests just the other day for develop on my local machine. The code passed just fine. Probably something changed on Travis.

Cheers,
Russell

On Aug 20, 2018, at 1:34 PM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


Soooo none of the downloads are working on travis, and I don't know why.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-414418515>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeQ4D1-fiH9dhhwOmjbB-h5VRCiRSks5uSwFHgaJpZM4VRgTW>.

aburrell commented:
I just tried to load a new day of data and it looks like the station reading functionality has broke.  Trying to fix it now.

rstoneback commented:
While I was poking around the build time issue I saw a complaint about it not liking {{:s}} as a format specifier for ints.

On Aug 24, 2018, at 9:47 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


I just tried to load a new day of data and it looks like the station reading functionality has broke. Trying to fix it now.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-415782102>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeayL03wnviy6zVcFCP3aelrL2_y9ks5uUBIJgaJpZM4VRgTW>.

aburrell commented:
That could be the reason why.

aburrell commented:
Ah, the empty test doesn't work because the files are being indexed by time incorrectly.  It's taking the closest available file instead of saying there's no file available.  Any idea where that could be happening?

aburrell commented:
Also, the closest available file was .csv instead of .txt so the format was all wrong.  It looks like the differentiation between multiple possible file types of the same data needs to be tested for.

rstoneback commented:
I'm a bit confused on the stored_file_info issue. I did find that
        if not files.empty:
            files.ix[files.index[-1] + doff -
                     pds.DateOffset(days=1)] = files.iloc[-1]
            files = files.asfreq('D', 'pad')
            # add the date to the filename
            files = files + '_' + files.index.strftime('%Y-%m-%d')

in the list_files routine will fill in days. Suppose I had a file for Jan 1, and a file for Dec 31. The snippet above will report files available through the whole year, even though there is only two.

Can you send me the commands that creates the issue you reported in #68?

aburrell commented:
I think it's just whatever command creates the stored_file_info files when you start pysat.  I responded more fully with attached files in an email.

aburrell commented:
I had to manually delete my stored_file_info.txt file, but it's working now.

rstoneback commented:
Good to hear, except the manual deletion part. Did you give it the update_files flag?

In [1]: import pysat

In [2]: mag = pysat.Instrument('supermag', 'magnetometer', tag='stations', update_files=True)
pysat is searching for supermag magnetometer stations files.
Found 731 of them.
When using this data please acknowledge the SuperMAG collaboration according to the request outlined in the metadata attribute 'acknowledgements'


On Sep 4, 2018, at 8:47 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


I had to manually delete my stored_file_info.txt file, but it's working now.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-418373322>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeVb-fkT6ARyzW0wmyYMDk-N3ucGbks5uXoR6gaJpZM4VRgTW>.

aburrell commented:
Nope, I couldn't remember what I was supposed to do :P

From: Russell Stoneback <notifications@github.com>
Reply-To: rstoneback/pysat <reply@reply.github.com>
Date: Tuesday, September 4, 2018 at 11:15
To: rstoneback/pysat <pysat@noreply.github.com>
Cc: "Burrell, Angeline" <angeline.burrell@nrl.navy.mil>, Mention <mention@noreply.github.com>
Subject: Re: [rstoneback/pysat] SuperMAG (#65)

Good to hear, except the manual deletion part. Did you give it the update_files flag?

In [1]: import pysat

In [2]: mag = pysat.Instrument('supermag', 'magnetometer', tag='stations', update_files=True)
pysat is searching for supermag magnetometer stations files.
Found 731 of them.
When using this data please acknowledge the SuperMAG collaboration according to the request outlined in the metadata attribute 'acknowledgements'


On Sep 4, 2018, at 8:47 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


I had to manually delete my stored_file_info.txt file, but it's working now.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-418373322>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeVb-fkT6ARyzW0wmyYMDk-N3ucGbks5uXoR6gaJpZM4VRgTW>.



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-418406718>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AGuC_jMGbXpFeFae_9-0aytWt6cu_Ek2ks5uXpkfgaJpZM4VRgTW>.

rstoneback commented:
No worries. I hear ya. That is the problem with options.

It is actually worse than the bunch of documented options in the Instrument constructor. There are also options that can be set at the instrument level that are engaged simply by setting a variable. It is fine if a person is fully up on python’s object model and they know it the option exists. There are notes in the documentation but these options don’t even have the benefit of a docstring.

Getting closer to old school Linux instructions. There isn’t a problem with ‘driver’, just set option blah in file you’ve never heard of, recompile the ‘driver manager’ and link back into the kernel.

Cheers,
Russell

On Sep 4, 2018, at 10:39 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:

Nope, I couldn't remember what I was supposed to do :P

From: Russell Stoneback <notifications@github.com<mailto:notifications@github.com>>
Reply-To: rstoneback/pysat <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, September 4, 2018 at 11:15
To: rstoneback/pysat <pysat@noreply.github.com<mailto:pysat@noreply.github.com>>
Cc: "Burrell, Angeline" <angeline.burrell@nrl.navy.mil<mailto:angeline.burrell@nrl.navy.mil>>, Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [rstoneback/pysat] SuperMAG (#65)

Good to hear, except the manual deletion part. Did you give it the update_files flag?

In [1]: import pysat

In [2]: mag = pysat.Instrument('supermag', 'magnetometer', tag='stations', update_files=True)
pysat is searching for supermag magnetometer stations files.
Found 731 of them.
When using this data please acknowledge the SuperMAG collaboration according to the request outlined in the metadata attribute 'acknowledgements'


On Sep 4, 2018, at 8:47 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


I had to manually delete my stored_file_info.txt file, but it's working now.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-418373322>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeVb-fkT6ARyzW0wmyYMDk-N3ucGbks5uXoR6gaJpZM4VRgTW>.



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-418406718>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AGuC_jMGbXpFeFae_9-0aytWt6cu_Ek2ks5uXpkfgaJpZM4VRgTW>.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/rstoneback/pysat/pull/65#issuecomment-418415590>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADbdeXrwGjMDyFJoe7KCHGflPsQeH83lks5uXp6bgaJpZM4VRgTW>.

aburrell commented:
Ok!  I looked at the travis-ci failure and this is where it's not working.  It's only not working in python 3.6.

> ======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer indices 
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 711, in download
    raise ValueError(estr)
ValueError: unable to format station query for [2009]
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): supermag.jhuapl.edu:80
urllib3.connectionpool: DEBUG: http://supermag.jhuapl.edu:80 "POST /mag/lib/services/??user=rstoneback&service=stations&fmt=ascii&year=2009 HTTP/1.1" 200 13222
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer  
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 711, in download
    raise ValueError(estr)
ValueError: unable to format station query for [2009]
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): supermag.jhuapl.edu:80
urllib3.connectionpool: DEBUG: http://supermag.jhuapl.edu:80 "POST /mag/lib/services/??user=rstoneback&service=stations&fmt=ascii&year=2009 HTTP/1.1" 200 13222
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: Checking download routine functionality for module:  supermag magnetometer all 
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 236, in check_download
    raise e
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 213, in check_download
    password=user_download_dict[inst_name][1])
  File "/home/travis/build/rstoneback/pysat/pysat/_instrument.py", line 967, in download
    password=password, **kwargs)
  File "/home/travis/build/rstoneback/pysat/pysat/instruments/supermag_magnetometer.py", line 711, in download
    raise ValueError(estr)
ValueError: unable to format station query for [2009]
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): supermag.jhuapl.edu:80
urllib3.connectionpool: DEBUG: http://supermag.jhuapl.edu:80 "POST /mag/lib/services/??user=rstoneback&service=stations&fmt=ascii&year=2009 HTTP/1.1" 200 13222
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: Checking load routine functionality for module with clean level "none":  supermag magnetometer stations 
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/rstoneback/pysat/pysat/tests/test_instruments.py", line 245, in check_load
    assert not inst.empty
AssertionError
